typing/typedecl.ml

```diff
@@ -263,54 +263,54 @@ let make_constructor env loc type_path type_params svars sargs sret_type =
   | Some sret_type ->
       (* if it's a generalized constructor we must first narrow and
          then widen so as to not introduce any new constraints *)
-      let z = narrow () in
+      (* narrow and widen are now invoked through wrap_type_variable_scope *)
+      with_local_type_variable_scope begin fun () ->
```

Comment:
@goldfirere: you mentioned that you looked at `Typetexp.{narrow,widen}`; it might be the change to a wrapper combinator proposed here conflicts with the refactoring you had in mind. (In that case, I would say submit anyway, we can worry about rebasing once something gets merged.)